src/class/HTTPCall/HTTPCall.cpp

```diff
@@ -154,6 +154,51 @@ int HTTPCall::getClientFd(void) const
 	return this->_client_fd;
 }
 
+Path HTTPCall::getPathServerDirectory(void) const
+{
+	Path serverPath(this->_virtual_server->getConf()->getRoot()->get());
```

Comment:
this is not so simple.
the path might be the path inside Location.
e.g.
```
server {
  root /var/www/html
  
  location /upload {
    root /tmp
  }
}
```

Reply:
i think it is best to move this logic to matcher namespace and have something like `Path matcher::rootToRequest(Server, Location, BasicHTTPRequest)`

inside, if the root is not in the location, then we take the one from server